src/EditorFeatures/TestUtilities/Async/Checkpoint.cs

```diff
@@ -11,11 +11,9 @@ namespace Roslyn.Test.Utilities
     public class Checkpoint
     {
         private readonly TaskCompletionSource<object> _tcs = new TaskCompletionSource<object>();
-        public Task Task { get { return _tcs.Task; } }
 
-        public void Release()
-        {
-            _tcs.TrySetResult(null);
-        }
+        public Task Task => _tcs.Task;
+
+        public bool TryRelease() => _tcs.TrySetResult(null);
```

Comment:
Agree with @jasonmalinowski.  To root out flaky behavior here make this type usage definitive.